app/services/post_alerter.rb

```diff
@@ -275,12 +275,17 @@ def category_or_tag_muters(topic)
       EXCEPT
       SELECT user_id FROM topic_users tus WHERE tus.topic_id = #{topic.id} AND tus.notification_level = #{TopicUser.notification_levels[:watching]}
     SQL
-    User
-      .where("id IN (#{user_ids_sql})")
-      .joins("LEFT JOIN user_options ON user_options.user_id = users.id")
-      .where(
-        "user_options.watched_precedence_over_muted IS false OR (user_options.watched_precedence_over_muted IS NULL AND #{!SiteSetting.watched_precedence_over_muted})",
+    user_scope = User.where("id IN (#{user_ids_sql})")
+
+    if SiteSetting.watched_precedence_over_muted
+      user_scope.joins(
+        "JOIN user_options ON user_options.user_id = users.id AND user_options.watched_precedence_over_muted IS false",
```

Comment:
If this becomes a INNER JOIN, we can union this result in the `user_ids_sql` subquery too.